src/demux/mp4demuxer.js

```diff
@@ -55,342 +54,40 @@ class MP4Demuxer {
-  }
-
-  static offsetStartDTS (initData, fragment, timeOffset) {
-    MP4Demuxer.findBox(fragment, ['moof', 'traf']).map(function (traf) {
-      return MP4Demuxer.findBox(traf, ['tfhd']).map(function (tfhd) {
-      // get the track id from the tfhd
-        let id = MP4Demuxer.readUint32(tfhd, 4);
-        // assume a 90kHz clock if no timescale was specified
-        let timescale = initData[id].timescale || 90e3;
-
-        // get the base media decode time from the tfdt
-        MP4Demuxer.findBox(traf, ['tfdt']).map(function (tfdt) {
-          let version = tfdt.data[tfdt.start];
-          let baseMediaDecodeTime = MP4Demuxer.readUint32(tfdt, 4);
-          if (version === 0) {
-            MP4Demuxer.writeUint32(tfdt, 4, baseMediaDecodeTime - timeOffset * timescale);
-          } else {
-            baseMediaDecodeTime *= Math.pow(2, 32);
-            baseMediaDecodeTime += MP4Demuxer.readUint32(tfdt, 8);
-            baseMediaDecodeTime -= timeOffset * timescale;
-            baseMediaDecodeTime = Math.max(baseMediaDecodeTime, 0);
-            const upper = Math.floor(baseMediaDecodeTime / (UINT32_MAX + 1));
-            const lower = Math.floor(baseMediaDecodeTime % (UINT32_MAX + 1));
-            MP4Demuxer.writeUint32(tfdt, 4, upper);
-            MP4Demuxer.writeUint32(tfdt, 8, lower);
-          }
-        });
-      });
-    });
-  }
-
   // feed incoming data to the front of the parsing pipeline
   append (data, timeOffset, contiguous, accurateTimeOffset) {
     let initData = this.initData;
+    const textTrack = {
+      samples: []
+    };
     if (!initData) {
       this.resetInitSegment(data, this.audioCodec, this.videoCodec, false);
       initData = this.initData;
     }
     let startDTS, initPTS = this.initPTS;
     if (initPTS === undefined) {
-      let startDTS = MP4Demuxer.getStartDTS(initData, data);
+      let startDTS = getStartDTS(initData, data);
       this.initPTS = initPTS = startDTS - timeOffset;
       this.observer.trigger(Event.INIT_PTS_FOUND, { initPTS: initPTS });
     }
-    MP4Demuxer.offsetStartDTS(initData, data, initPTS);
-    startDTS = MP4Demuxer.getStartDTS(initData, data);
-    this.remuxer.remux(initData.audio, initData.video, null, null, startDTS, contiguous, accurateTimeOffset, data);
+    if (initData && initData[1] && initData[1].type === 'video') {
+      // parse shit here
+      const trackId = initData.video && initData.video.id;
+      const timescale = initData.video && initData.video.timescale;
+      if (trackId && timescale) {
+        textTrack.samples = parseTextTrackSamplesFromVideoSegment(data, trackId);
+        textTrack.timescale = timescale;
+        textTrack.initPTS = this.initPTS;
+      }
+    }
+    offsetStartDTS(initData, data, initPTS);
+    startDTS = getStartDTS(initData, data);
+    this.remuxer.remux(initData.audio, initData.video, null, textTrack, startDTS, contiguous, accurateTimeOffset, data);
+  }
+
+  static probe (data) {
```

Comment:
because tthey attach demuxer probe to n
`      for (let i = 0, len = muxConfig.length; i < len; i++) {
        const mux = muxConfig[i];
        const probe = mux.demux.probe;
        if (probe(data)) {
          const remuxer = this.remuxer = new mux.remux(observer, config, typeSupported, this.vendor);
          demuxer = new mux.demux(observer, remuxer, config, typeSupported);
          this.probe = probe;
          break;
        }
      }`

Reply:
Works for me